.gitignore

```diff
@@ -29,6 +29,7 @@ dlangspec.tex
 
 /web
 /2.*.ddoc
+modlist-*.ddoc
```

Comment:
> Who generates these files

`modlist.d`, added in this PR

> and where should they go at best?

These are intermediary source files.

What is a problem with a long .gitignore? It is not an issue with many projects. You can use `git clean` to clear detrius.

Temporary files *should* be somewhere under the repository's directory. Otherwise, if the `clean` target malfunctions, the user has no idea which files he has to delete. I've ran into this problem just this week, with Dub and its `~/.dub` directory.

I suppose we could put temporary files in a `tmp` subdirectory, but it's a trade-off between Makefile complexity and a shorter `.gitignore`, which IMHO is no contest.